base/MAPL_Generic.F90

```diff
@@ -441,6 +445,10 @@ module MAPL_GenericMod
    type (MAPL_Communicators)                :: mapl_comm
    type (TimeProfiler), public :: t_profiler
    character(:), allocatable :: full_name ! Period separated list of ancestor names
+
+   type(MAPL_VarServiceProviderPtr), pointer :: provider_list(:) => null()
+   type(MAPL_VarServiceSubscriberPtr), pointer :: subscriber_list(:) => null()
+
```

Comment:
This might be a good place to introduce Vector containers to support the growable list - depends on what I see about their use below.